feat: Improve path cleaning UI/UX

rafaeelaudibert opened the pull request:
Paths displayed on Web Analytics/Web Vitals table now properly render them using the new readable format for them, assuming people use the suggested `<slug>` format

![image](https://github.com/user-attachments/assets/47347c22-79b1-4ae9-8717-ceca00b537dd)
![image](https://github.com/user-attachments/assets/0b01aa14-1f1a-4413-bdf1-d6a113a4f3a6)

I've also made path cleaning a top-level filter on Web Analytics because I want to push people to use it to increase adoption. Once people have them set-up, the toolbar will be a no-brainer to explore Web Vitals

![image](https://github.com/user-attachments/assets/0c4df3f1-d17c-4fad-947b-28a33378511b)
![image](https://github.com/user-attachments/assets/b774c967-2e84-4c7b-a0d5-11462c2d1a57)

We've also changed how filtering works so that if someone has path cleaning enabled and clicks in one of the cleaned paths, we'll properly display results for that filter 🎉. To make things even, we've added this feature to Web Vitals as well.

rafaeelaudibert committed:
refactor(web-analytics): Move live count to the top

rafaeelaudibert committed:
feat: Make path cleaning first-class citizen in the filters list

Ideally we'd want a smaller button, but people aren't familar with path cleaning yet, so let's make it slightly exaggerated

rafaeelaudibert committed:
feat(): Improve path cleaning UI

This is now much cleaner, and we also have a way to test what the path will look like once cleaned

rafaeelaudibert committed:
feat: Highlight path cleaning on Web Analytics/Web Vitals

Paths displayed on Web Analytics/Web Vitals table now properly render them using the new readable format for them, assuming people use the suggested `<slug>` format

In a future PR we'll enable people to filter by cleaned paths, making cleaned paths much more powerful for WA/WV

rafaeelaudibert commented:
@robbie-c In case you're wondering where I'm going with this, I need the capability to, when looking from the toolbar, see all of the events from that page after it's been aliased, so I need whatever I'm implementing in the following PR, this one is just cleaning this up and making the feature slightly more useful

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

rafaeelaudibert committed:
fix: Prevent users from toggling path cleaning if feature is not enabled

rafaeelaudibert committed:
feat: Allow Web Analytics to filter by cleaned path

We're now allowing us to filter by cleaned path, which makes our UI slightly more powerful. If you select a cleaned path and `doPathCleaning` is set, we can see results from a specific cleaned path. The UX isn't great because it looks kinda wonky, and I'm pretending it's a `EXACT` match (which kinda is?) and you can't get into that state unless you click on a path in the UI, but it does work, so I think this is a win. Eventually once we have `$pathname` a high-level filter (rather than just another filter) we'll be able to make this behave better.

Web Vitals is coming in the next commit

rafaeelaudibert committed:
feat: Make search work in the opposite direction

Now, if path cleaning is enabled and I filter by a path that would be
cleaned, we display data from the filtered path instead

rafaeelaudibert committed:
feat: Allow paths to be filtered on web vitals path breakdown

You can now click on a path on web vitals path breakdown to add it to the filter

rafaeelaudibert committed:
feat: Detect `:id` as a valid alias

rafaeelaudibert committed:
refactor: Simplify unit calculation

We don't need to calculate that before hand, we can simply pass the metric to the function and it knows how to handle it

rafaeelaudibert commented:
Thanks for merging with master @robbie-c, would love a review :)

robbie-c reviewed (commented):
Apologies that this is a partial review - will finish the review tomorrow morning my time

rafaeelaudibert commented on posthog/hogql/property.py:
```diff
@@ -234,10 +234,33 @@ def _expr_to_compare_op(
         return ast.CompareOperation(op=ast.CompareOperationOp.LtEq, left=expr, right=ast.Constant(value=value))
     elif operator == PropertyOperator.GTE:
         return ast.CompareOperation(op=ast.CompareOperationOp.GtEq, left=expr, right=ast.Constant(value=value))
+    elif operator == PropertyOperator.IS_CLEANED_PATH_EXACT:
```
The thing is, Im not adding it as an operator that can be selected in the UI at the moment, Im dynamically translating it to this operator when needed.

I see the benefit to creating them "for real" eventually, but it's not something we need now.

I couldnt come up with a more elegant solution, any ideas?

robbie-c replied:
Ah if it's not in the UI then I'm less worried, that makes sense

robbie-c commented on frontend/src/scenes/web-analytics/webAnalyticsLogic.tsx:
```diff
@@ -689,7 +706,7 @@ export const webAnalyticsLogic = kea<webAnalyticsLogicType>([
                     : undefined
 
-                // the queries don't currently revenue when the conversion goal is an action
+                // the queries don't currently include revenue when the conversion goal is an action
```
🙃

robbie-c reviewed (commented):
Broadly, this looks great! I approve, and props for improving this.

I haven't had a chance to run this locally yet - but one thing I am concerned about is backwards compatability. I would like to see something that shows that this has been considered. Either:
* put it behind a FF so we can roll out slowly and revert if a customer's use case breaks
* write some tests to show that backwards compat is fine
* write a big comment explaining why everything is backwards compatible

robbie-c reviewed (approved):
Approved but add a FF :)

rafaeelaudibert committed:
feat: Add feature flag to toggle path cleaning filter off

We don't believe we'll need to disable this in the future, but let's keep this around as a safety switch

rafaeelaudibert commented:
FF added on [16c0621](https://github.com/PostHog/posthog/pull/27997/commits/16c0621d9dfc38da25ce80ab54b2de496862cf1c), will merge this with 100% turned on, will act as a kill-switch, thanks!

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)
